standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestGetAllTableConstraints.java

```diff
@@ -0,0 +1,382 @@
+
+    // Assert Not Null constraint
+    Assert.assertEquals(1, fetched.getNotNullConstraintsSize());
+    Assert.assertEquals(table.getDbName(), fetched.getNotNullConstraints().get(0).getTable_db());
+    Assert.assertEquals(table.getTableName(), fetched.getNotNullConstraints().get(0).getTable_name());
+    Assert.assertEquals("col3", fetched.getNotNullConstraints().get(0).getColumn_name());
+    Assert.assertEquals(nnCol3ConstraintName, fetched.getNotNullConstraints().get(0).getNn_name());
+    Assert.assertTrue(fetched.getNotNullConstraints().get(0).isEnable_cstr());
+    Assert.assertFalse(fetched.getNotNullConstraints().get(0).isValidate_cstr());
+    Assert.assertFalse(fetched.getNotNullConstraints().get(0).isRely_cstr());
+    Assert.assertEquals(table.getCatName(), fetched.getNotNullConstraints().get(0).getCatName());
+
+    // Assert check constraint
+    Assert.assertEquals(1, fetched.getNotNullConstraintsSize());
+    Assert.assertEquals(table.getDbName(), fetched.getCheckConstraints().get(0).getTable_db());
+    Assert.assertEquals(table.getTableName(), fetched.getCheckConstraints().get(0).getTable_name());
+    Assert.assertEquals("col2", fetched.getCheckConstraints().get(0).getColumn_name());
+    Assert.assertEquals("= 5", fetched.getCheckConstraints().get(0).getCheck_expression());
+    Assert.assertEquals(ccCol2ConstraintName, fetched.getCheckConstraints().get(0).getDc_name());
+    Assert.assertTrue(fetched.getCheckConstraints().get(0).isEnable_cstr());
+    Assert.assertFalse(fetched.getCheckConstraints().get(0).isValidate_cstr());
+    Assert.assertFalse(fetched.getCheckConstraints().get(0).isRely_cstr());
+    Assert.assertEquals(table.getCatName(), fetched.getCheckConstraints().get(0).getCatName());
+
+    // Assert foreign key
+    Assert.assertEquals(1, fetched.getForeignKeysSize());
+    Assert.assertEquals(table.getDbName(), fetched.getForeignKeys().get(0).getFktable_db());
+    Assert.assertEquals(table.getTableName(), fetched.getForeignKeys().get(0).getFktable_name());
+    Assert.assertEquals("col4", fetched.getForeignKeys().get(0).getFkcolumn_name());
+    Assert.assertEquals(parentTable.getDbName(), fetched.getForeignKeys().get(0).getPktable_db());
+    Assert.assertEquals(parentTable.getTableName(), fetched.getForeignKeys().get(0).getPktable_name());
+    Assert.assertEquals("col1", fetched.getForeignKeys().get(0).getPkcolumn_name());
+    Assert.assertEquals(1, fetched.getForeignKeys().get(0).getKey_seq());
+    Assert.assertEquals(parentPkConstraintName, fetched.getForeignKeys().get(0).getPk_name());
+    Assert.assertEquals(fkConstraintName, fetched.getForeignKeys().get(0).getFk_name());
+    Assert.assertTrue(fetched.getForeignKeys().get(0).isEnable_cstr());
+    Assert.assertFalse(fetched.getForeignKeys().get(0).isValidate_cstr());
+    Assert.assertFalse(fetched.getForeignKeys().get(0).isRely_cstr());
+    Assert.assertEquals(table.getCatName(), fetched.getForeignKeys().get(0).getCatName());
+
+  }
```

Comment:
Add a negative scenario where db invalid or table invalid etc and check if we get NoSuchObjectException.

Reply:
We will not get NoSuchObjectException because in the code follow NoSuchObjectException is replaced by MetaException